### ISSUE TITLE
Feature request: AS_EMPTY as DeserializationFeature

### ISSUE THREAD

**Issue by jlous** [FasterXML/jackson-databind#3645]
I wish I could configure my object mapper to use "Nulls.AS_EMPTY" as default for deserializing list fields.

* When I design with non-nullable lists, I commonly do so as a general design principle for the entire model, not just particular fields
* It is sometimes desirable to use the same Pojos for multipe integrations with different requirements.  It is therefore good to be able to configure (de)serialisation on a per-integration object mapper level rather than globally on the model.
* I generally like to keep my domain classes as "clean" as possible

The perfect solution for me would automatically adjust to whether my kotlin field was nullable or not, but a fixed policy will do the job for most cases.

**Comment by pjfanning** [comment]
Is this Kotlin specific? If so, could you move this the jackson-module-kotlin issues list?

**Comment by jlous** [comment]
Not really. Only the extra wish at the end.  The deserializationFeature would be equally relevant in Java.

**Comment by cowtowncoder** [comment]
Ok yes, I think this is the right place for the issue.

This would not belong under `DeserializationFeature` since it's too granular. 
But I think it could work as "config override" per type, although would have to be considered at general `List.class` level. And would not just be for `List`s or just "as Empty" so would have wider potential usability.

**Comment by cowtowncoder** [comment]
```Can we consider creating collection-specific feature? something like CollectionFeature. READ_NULL_TO_EMPTY?
Like we have for EnumFeature?
```

Unfortunately no, as I don't like having features that overlap with other configuration functionality, and `ConfigOverride` is the main mechanism (along with `@JsonSetter` annotation) for applying alternate Null handling. It exists, is used; and if we had different way of configuring, we would have to figure out how to merge different "dimensions" of configurability.

`ConfigOverride` has the challenge of how to apply scopes as target type must be indicated with `Class`.
I will note that one can apply definition for `List`, alternatively; question then is just whether it's possible to support multiple levels (that is, if no definition for `List`, then check for `Collection` -- this is not used by any of existing deserializers).
